security/self-assessment.md

```diff
@@ -0,0 +1,495 @@
+- [Security links](#security-links)
+- [Software Bill of Materials](#software-bill-of-materials)
+- [Overview](#overview)
+  - [Background](#background)
+  - [Actors](#actors)
+  - [Actions](#actions)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Self-assessment Use](#self-assessment-use)
+- [Security functions and features](#security-functions-and-features)
+- [Project Compliance](#project-compliance)
+- [Secure Development Practices](#secure-development-practices)
+  - [Development pipeline](#development-pipeline)
+  - [Communication Channels](#communication-channels)
+  - [Ecosystem](#ecosystem)
+- [Security issue resolution](#security-issue-resolution)
+  - [Responsible disclosure practice](#responsible-disclosure-practice)
+  - [Incident response](#incident-response)
+- [Appendix](#appendix)
+
+# Metadata
+
+<table>
+  <thead>
+    <tr>
+      <th>Name</th>
+      <th>Information</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Assessment Stage</td>
+      <td>Incomplete</td>
+    </tr>
+    <tr>
+      <td>Software</td>
+      <td>
+        <a href="https://github.com/kubeflow/spark-operator">Kubeflow Spark Operator</a>, 
+        <a href="https://github.com/kubeflow/notebooks">Kubeflow Notebooks</a>, 
+        <a href="https://github.com/kubeflow/trainer">Kubeflow Trainer</a>, 
+        <a href="https://github.com/kubeflow/katib">Kubeflow Katib</a>, 
+        <a href="https://github.com/kubeflow/model-registry">Kubeflow Model Registry</a>, 
+        <a href="https://github.com/kubeflow/pipelines">Kubeflow Pipelines</a>
+      </td>
+    </tr>
+    <tr>
+      <td>Security Provider?</td>
+      <td>No. Kubeflow projects are not security providers</td>
+    </tr>
+    <tr>
+      <td>Languages</td>
+      <td>Python, Go, TypeScript</td>
+    </tr>
+  </tbody>
+</table>
+
+# Security links
+
+- Kubeflow Spark Operator security policy: https://github.com/kubeflow/spark-operator/blob/master/SECURITY.md
+- Kubeflow Notebooks security policy: https://github.com/kubeflow/notebooks/blob/master/SECURITY.md
+- Kubeflow Trainer security policy: https://github.com/kubeflow/trainer/blob/master/SECURITY.md
+- Kubeflow Katib security policy: https://github.com/kubeflow/katib/blob/master/SECURITY.md
+- Kubeflow Model Registry security policy: https://github.com/kubeflow/model-registry/blob/main/SECURITY.md
+- Kubeflow Pipelines security policy: https://github.com/kubeflow/pipelines/blob/master/SECURITY.md
+
+# Software Bill of Materials
+
+The go.mod file for each component can be found in the appropriate Kubeflow projects:
+
+- https://github.com/kubeflow/spark-operator/blob/master/go.mod
+- https://github.com/kubeflow/notebooks/blob/notebooks-v2/workspaces/controller/go.mod
+- https://github.com/kubeflow/trainer/blob/master/go.mod
+- https://github.com/kubeflow/katib/blob/master/go.mod
+- https://github.com/kubeflow/model-registry/blob/main/go.mod, https://github.com/kubeflow/model-registry/blob/main/clients/python/pyproject.toml
```

Comment:
Does it matter than https://github.com/kubeflow/model-registry/blob/main/clients/python/pyproject.toml is not a go.mod file?

Reply:
Ideally, we should also keep packages for SDK, UIs, and controllers.

Reply:
Oh, you are right. Let me try to group them by project.